app/controllers/application_controller.rb

```diff
@@ -4,7 +4,7 @@ class ApplicationController < ActionController::Base
 
   private
   def set_locale
-    locale = params[:locale].to_s.strip.to_sym
+    locale = params.dig(:locale).to_s.strip.to_sym
```

Comment:
Tại sao chỗ này lại cần dùng dig em nhỉ

Reply:
Em đang hiểu là chỗ nào cũng có thể áp dụng dig thay vì truy cập thẳng vì nó đang an toàn hơn ạ

Reply:
Trong trường hợp của em thì nó ko khác gì gọi trực tiếp cả nhé. Thằng dig nó được sử dụng khi em muốn truy cập sâu vào các nested value nhé.
VD: params = { user: { class: 5}}.
Để lấy được giá trị 5 có 2 cách: 
- C1: params[:user][:class]
- C2: params.dig(:user, :class).
Thay đổi key ban đầu thành: params = { grade: {class: 5}}.
Giờ nếu gọi params[:user][:class] thì sẽ raise exception, trong khi gọi params.dig(:user, :class) thì trả về nil.
Làm vậy để tránh gây lỗi khi đang sử dụng nhé. Nó cx tương tự như dùng find_by thay vì dùng find.

Reply:
à em hiểu rồi ạ